### PR TITLE
service: show warning on missing task callbacks

### PR DESCRIPTION
**Pull request by ilgooz**
closes https://github.com/mesg-foundation/mesg-js/issues/79

### CODE REVIEW

File: src/service/service.ts [hunk 1]
```diff
@@ -70,12 +70,12 @@ class Service {
...
     const nonDescribedTasks = Object.keys(this.tasks).filter(x => !this.definition.tasks[x]);
-    if (nonDescribedTasks.length > 0){
+    if (nonDescribedTasks.length > 0) {
       throw new Error(`The following tasks does not present in your mesg.yml: ${nonDescribedTasks.join(', ')}`);
     }
     const nonHandledTasks = Object.keys(this.definition.tasks).filter(x => !this.tasks[x]);
-    if (nonHandledTasks.length > 0){
-      throw new Error(`The following tasks described in your mesg.yml don't have callbacks: ${nonHandledTasks.join(', ')}`);
+    if (nonHandledTasks.length > 0) {
+      console.log(`WARNING: The following tasks described in your mesg.yml don't have callbacks: ${nonHandledTasks.join(', ')}`);
```
Comment (antho1404) [inline src/service/service.ts]: Talking about callback is confusing I think.

```suggestion
      console.log(`WARNING: The following tasks described in your mesg.yml are not implemented: ${nonHandledTasks.join(', ')}`);
```

File: src/service/service.ts [hunk 2]
```diff
@@ -70,12 +70,12 @@ class Service {
 
   private validateTaskNames(){
     const nonDescribedTasks = Object.keys(this.tasks).filter(x => !this.definition.tasks[x]);
-    if (nonDescribedTasks.length > 0){
+    if (nonDescribedTasks.length > 0) {
       throw new Error(`The following tasks does not present in your mesg.yml: ${nonDescribedTasks.join(', ')}`);
```
Comment (NicolasMahe) [inline src/service/service.ts]:
```suggestion
      throw new Error(`The following tasks are not present in the mesg.yml: ${nonDescribedTasks.join(', ')}`);
```

File: src/service/service.ts (hunk 1 above)
Comment (ilgooz) [reply src/service/service.ts]: done

File: src/service/service.ts (hunk 2 above)
Comment (ilgooz) [reply src/service/service.ts]: done

File: src/service/service.ts [hunk 3]
```diff
@@ -71,12 +71,12 @@ class Service {
...
-    if (nonDescribedTasks.length > 0){
-      throw new Error(`The following tasks does not present in your mesg.yml: ${nonDescribedTasks.join(', ')}`);
+    if (nonDescribedTasks.length > 0) {
+      throw new Error(`The following tasks don't present in the mesg.yml: ${nonDescribedTasks.join(', ')}`);
     }
     const nonHandledTasks = Object.keys(this.definition.tasks).filter(x => !this.tasks[x]);
-    if (nonHandledTasks.length > 0){
-      throw new Error(`The following tasks described in your mesg.yml don't have callbacks: ${nonHandledTasks.join(', ')}`);
+    if (nonHandledTasks.length > 0) {
+      console.log(`WARNING: The following tasks described in the mesg.yml haven't implemented: ${nonHandledTasks.join(', ')}`);
```
Comment (antho1404) [inline src/service/service.ts]:
```suggestion
      console.log(`WARNING: The following tasks described in the mesg.yml haven't been implemented: ${nonHandledTasks.join(', ')}`);
```

File: src/service/service_test.ts [hunk 4]
```diff
@@ -56,17 +56,15 @@ test('listenTask() should throw because missing task in mesg.yml', function (t)
...
   const service = newService(definition, client);
-  try {
-    service.listenTask({ "task1": () => {} });
-    t.fail("should throw");
-  } catch(e) {
-    t.ok(e);
-  }
+  const spy = sinon.spy(console, 'log');
+  service.listenTask({ "task1": () => {} });
+  t.equal(spy.getCall(0).args[0], 'WARNING: The following tasks described in the mesg.yml haven\'t implemented: task2');
```
Comment (antho1404) [inline src/service/service_test.ts]:
```suggestion
  t.equal(spy.getCall(0).args[0], 'WARNING: The following tasks described in the mesg.yml haven\'t been implemented: task2');
```

File: src/service/service.ts [hunk 5]
```diff
@@ -71,12 +71,12 @@ class Service {
 
   private validateTaskNames(){
     const nonDescribedTasks = Object.keys(this.tasks).filter(x => !this.definition.tasks[x]);
-    if (nonDescribedTasks.length > 0){
-      throw new Error(`The following tasks does not present in your mesg.yml: ${nonDescribedTasks.join(', ')}`);
+    if (nonDescribedTasks.length > 0) {
+      throw new Error(`The following tasks don't present in the mesg.yml: ${nonDescribedTasks.join(', ')}`);
```
Comment (NicolasMahe) [inline src/service/service.ts]:
```suggestion
      throw new Error(`The following tasks are not present in the mesg.yml: ${nonDescribedTasks.join(', ')}`);
```

File: src/service/service.ts [hunk 6]
```diff
@@ -71,12 +71,12 @@ class Service {
...
     }
     const nonHandledTasks = Object.keys(this.definition.tasks).filter(x => !this.tasks[x]);
-    if (nonHandledTasks.length > 0){
-      throw new Error(`The following tasks described in your mesg.yml don't have callbacks: ${nonHandledTasks.join(', ')}`);
+    if (nonHandledTasks.length > 0) {
+      console.log(`WARNING: The following tasks described in the mesg.yml haven't been implemented: ${nonHandledTasks.join(', ')}`);
```
Comment (NicolasMahe) [inline src/service/service.ts]:
```suggestion
      console.warn(`WARNING: The following tasks described in the mesg.yml haven't been implemented: ${nonHandledTasks.join(', ')}`);
```

File: src/service/service_test.ts (hunk 4 above)
Comment (ilgooz) [reply src/service/service_test.ts]: done

**Pull request merged**